src/openms_gui/source/VISUAL/DIALOGS/ToolsDialog.cpp

```diff
@@ -440,4 +440,19 @@ namespace OpenMS
     return tools_combo_->currentText();
   }
 
+  String ToolsDialog::getExtension()
+  {
+    // Try to Return the first valid string for the extension on the output parameter
+    String extension = FileTypes::typeToName(FileTypes::UNKNOWN);
+    try 
+    {
+      extension = vis_param_.getValidStrings(output_combo_->currentText().toStdString())[0];
+    }
+    catch (Exception::BaseException e)
+    {
+      QMessageBox::critical(this, "Error", QString("Error storing INI file: ") + e.what());
```

Comment:
I think this needs some thought.
This will crash TOPPView if an empty vector is returned (try/catch will not help)